docker/docker_image.go

```diff
@@ -172,10 +186,39 @@ func sanitize(s string) string {
 }
 
 func (i *dockerImage) getSchema1Manifest() (manifest, error) {
-	manblob, err := i.Manifest()
+	manblob, ct, err := i.Manifest()
 	if err != nil {
 		return nil, err
 	}
+	switch ct {
+	case utils.V2ListMIMEType:
+		var digest string
+		ml := &manifestList{}
+		if err := json.Unmarshal(manblob, ml); err != nil {
+			return nil, err
+		}
+		// TODO(provide a flag/arg to select which os/arch to use)
+		for _, m := range ml.Manifests {
+			if m.Platform.Architecture == runtime.GOARCH && m.Platform.Os == runtime.GOOS {
+				digest = m.Digest
+				break
+			}
+		}
+		if digest == "" {
+			return nil, errors.New("no supported platform found in manifest list")
+		}
+		url := fmt.Sprintf(manifestURL, i.src.ref.RemoteName(), digest)
+		// just Accept v2s1 by not providing Accept headers here
+		res, err := i.src.c.makeRequest("GET", url, nil, nil)
```

Comment:
I think it would be clean to add a `desiredMIMEType` (with `""` meaning “default/best available”) to `ImageSource.GetManifest`.